OpenRA.Mods.Common/Activities/FindResources.cs

```diff
@@ -51,7 +51,7 @@ public FindResources(Actor self, CPos avoidCell)
 
 		public override Activity Tick(Actor self)
 		{
-			if (IsCanceled || NextActivity != null)
+			if (IsCanceled || NextInQueue != null)
 				return NextActivity;
```

Comment:
> Why can't we check NextActivity != null in this case?

Because `NextActivity` will be non-null as soon as there is a parent, which would then immediately quit the activity.

> Is there a situation where NextActivity can be null when NextInQueue isn't null?

No.

> Splitting this into two checks might make this clearer?

Yup, will do.